assets/src/edit-story/app/media/utils/getResourceFromLocalFile.js

```diff
@@ -120,21 +129,48 @@ const getVideoResource = async (file) => {
   });
 };
 
+const createPlaceholderResource = (properties) => {
+  return createLocalResource({ ...properties, isPlaceholder: true });
+};
+
+const getPlaceholderResource = (file) => {
+  const type = getTypeFromMime(file.type);
+  const mimeType = type === 'image' ? 'image/png' : 'video/mp4';
+
+  // The media library requires resources with valid mimeType and dimensions.
+  return createPlaceholderResource({
+    type: type || 'image',
+    mimeType: mimeType,
+    src: '',
```

Comment:
Would the source here be an blob of transparent 1x1 pixel?

Reply:
Placeholders were implemented in a way that they explicitly do not have a source.

Resources without a source (placeholders) won't have any video playback button, no poster generation, etc. They're just "blank".